build_runner_core/lib/src/util/constants.dart

```diff
@@ -18,7 +18,7 @@ String assetGraphPathFor(String path) =>
 /// Relative path to the directory which holds serialized versions of errors
 /// reported during previous builds.
 final errorCachePath =
-    p.url.join(cacheDir, _scriptHashFor(_scriptPath), 'error_cache');
+    p.join(cacheDir, _scriptHashFor(_scriptPath), 'error_cache');
```

Comment:
`p.posix`?

Reply:
This one can be the host OS style since it's only going to flow through to another OS style `p.join`. The only existing usage was round tripping between `p.url.join` then `p.fromUrl`, so I'm keeping it OS style the whole way instead.